generators/inverter_dyn.cpp

```diff
@@ -6094,8 +6108,16 @@ STATUS inverter_dyn::updateCurrInjection(int64 iteration_count,bool *converged_f
 					//Compare it
 					if ((terminal_current_val_pu[loop_var].Mag() > Imax) && running_in_delta)	//Current limit only gets applied when controls valid (deltamode)
 					{
+					  // Calculate phase angle correction
+					  double theta = terminal_current_val_pu[loop_var].Arg(); // Current angle
```

Comment:
@WeiDu276 : Please check if this code for angle correction matches with what you have in your paper.